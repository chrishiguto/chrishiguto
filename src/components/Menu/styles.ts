import styled from 'styled-components'

export const Wrapper = styled.menu`
  display: flex;
  align-items: center;
  width: 100%;
  height: 7.2rem;
`
