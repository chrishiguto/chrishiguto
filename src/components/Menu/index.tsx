import { Sun } from '@styled-icons/feather/Sun'
import { Moon } from '@styled-icons/feather/Moon'

import { useTheme, ThemeProps } from 'hooks/theme'

import { Container } from 'components/Container'
import Heading from 'components/Heading'
import Switch from 'components/Switch'

import * as S from './styles'

const Menu = () => {
  const { toggleTheme, userTheme } = useTheme()

  return (
    <S.Wrapper>
      <Container>
        <S.Menu>
          <Heading size="large">ch.</Heading>
          {!!userTheme && (
            <Switch
              activeIcon={<Moon size={16} strokeWidth={2} />}
              inactiveIcon={<Sun size={16} strokeWidth={2} />}
              checked={userTheme === 'light'}
              handleClick={toggleTheme}
            />
          )}
        </S.Menu>
      </Container>
    </S.Wrapper>
  )
}

export default Menu
