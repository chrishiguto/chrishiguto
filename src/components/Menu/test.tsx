import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Menu from '.'

describe('<Menu />', () => {
  it('should render the menu', () => {
    const { container } = renderWithTheme(<Menu />)

    expect(screen.getByRole('heading', { name: /ch\./i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
