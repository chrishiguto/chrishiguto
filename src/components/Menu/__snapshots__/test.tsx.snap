// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should render the menu 1`] = `
.c1 {
  width: 100%;
  max-width: 130rem;
  margin-left: auto;
  margin-right: auto;
  padding-left: calc(3.2rem / 2);
  padding-right: calc(3.2rem / 2);
}

.c3 {
  font-size: 1.8rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 7.2rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<header
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h2
        class="c3"
      >
        ch.
      </h2>
    </div>
  </div>
</header>
`;
